tests/unix_listener.rs

```diff
@@ -89,6 +89,7 @@ fn unix_listener_register() {
 }
 
 #[test]
+#[cfg_attr(target_os = "cygwin", ignore = "Cygwin blocks on connect")]
```

Comment:
We need to fix this, otherwise Mio won't work properly.

Reply:
Welcome to join the discussion, then:)

https://inbox.sourceware.org/cygwin/Z_UERXFI1g-1v3p2@calimero.vinschen.de/T/#t

TL;DR: They provide a workaround, which sacrifices `PEERCRED`; and they don't think it's a bug.

Reply:
Well I think it's cygwin's problem, and mio doesn't have to try to fix it. Cygwin target is only tier-3, which means anyone can practically break it without any doubt. This PR just wants to make it compile and "nearly" work on cygwin (and it really works well except unix sockets). If there is any person complaining about the bug or misbehavior of mio on cygwin, just assign it to me:) There won't be many people concern about this target, though.